packages/frontend/amp/lib/find-adslots.ts

```diff
@@ -72,68 +74,92 @@ const hasForwardBuffer = (
     elements: ElementWithLength[],
     index: number,
 ): boolean => {
-    const forwardElements = elements.slice(-1 * index);
+    const forwardElements = elements.slice(index + 1, elements.length);
     const meetsThreshold =
-        getLengthOfFollowingTextElements(forwardElements) >= IMG_BUFFER_FWD;
+        getLengthOfFollowingTextElements(forwardElements) >=
+        NONTEXT_BUFFER_FORWARD;
     const noForwardsEmbeds =
         forwardElements.filter(e => isTextElement(e.element)).length ===
         forwardElements.length;
 
     const enoughCharsForward = meetsThreshold || noForwardsEmbeds;
 
-    const neighbourSuitable = elements[index + 1]
-        ? suitableAdNeighbour(elements[index + 1])
+    const neighbourSuitable = elements[index]
+        ? suitableAdNeighbour(elements[index])
         : false;
     return enoughCharsForward && neighbourSuitable;
 };
 
 const hasBackwardBuffer = (
     elements: ElementWithLength[],
     index: number,
-    textSinceLastAd: number,
 ): boolean => {
-    const backwardsElements = elements.slice(0, index + 1).reverse();
+    const backwardsElements = elements.slice(0, index).reverse();
     const meetsThreshold =
-        getLengthOfFollowingTextElements(backwardsElements) >= IMG_BUFFER_BWD;
+        getLengthOfFollowingTextElements(backwardsElements) >=
+        NONTEXT_BUFFER_BACKWARD;
     const noBackwardsEmbeds =
         backwardsElements.filter(e => isTextElement(e.element)).length ===
         backwardsElements.length;
 
     const enoughCharsBackward = meetsThreshold || noBackwardsEmbeds;
 
+    return suitableAdNeighbour(elements[index]) && enoughCharsBackward;
+};
+
+/**
+ * There are some magic numbers here - obtained by eyeballing content. Feel free
+ * to update them if you notice issues.
+ */
+const sufficientSpaceSinceLastAd = (
```

Comment:
Should this be moved to within `hasBackwardBuffer`? It feels like they overlap in purpose a bit?